### ISSUE TITLE
Who decides subtitles on playback start broken?

### ISSUE THREAD

**Issue by mason1171** [croneter/PlexKodiConnect#1759]
## Help yourself
* I did try to restart Kodi :-)
* I checked the [PKC Frequently Asked Questions on the PKC wiki](https://github.com/croneter/PlexKodiConnect/wiki/faq)
* I did try to reset the Kodi database by going to `PKC Settings -> Advanced -> "Reset the database and optionally reset PlexKodiConnect"` and then hitting YES, NO
* I did check the [existing issues on Github](https://github.com/croneter/PlexKodiConnect/issues)

## Describe the bug
Before 3.6.2, the Who decides subtitles on playback start option (Plex) worked perfectly. After the update, the feature is broken.

## To Reproduce
1. Open the PKC settings
2. Set the previously mentioned option to Plex
3. Find a file with more than one subtitle. 
4. If the file’s default subtitles differs from plex’s preferences, PKC will likely ignore the proper choice.

## Expected behavior
Ex. If the file has Japanese subtitles (default) alongside English subtitles and Plex preferences list English subtitles as preferred, PKC should switch to English subtitles

The Who decides audio on playback start option does still work luckily.

## You need to attach a KODI LOG FILE!
Log was too big for pastebin
https://we.tl/t-MwvjlyxquE

I am aware that I can delete Plex tokens that I accidentially posted by following the [instructions on the PKC wiki](https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug#i-published-my-plex-token-to-some-forum-or-github-anyone-can-now-access-my-plex-server)

**Comment by croneter** [comment]
Your log looks just fine:
```
2021-12-28 00:58:20.965 T:4055120   DEBUG <general>: PLEX.subtitles: Kodi list of available subtitles: ['eng', 'eng']
2021-12-28 00:58:20.965 T:4055120   DEBUG <general>: PLEX.subtitles: Available Plex subtitle streams for currently playing item:
2021-12-28 00:58:20.965 T:4055120   DEBUG <general>: PLEX.subtitles: Number 0: Stream: {'id': '188774', 'streamType': '3', 'default': '1', 'codec': 'pgs', 'index': '3', 'bitrate': '6', 'language': 'English', 'languageTag': 'en', 'languageCode': 'eng', 'title': 'Signs / Songs', 'displayTitle': 'English (PGS)', 'extendedDisplayTitle': 'Signs / Songs (English PGS)'}
2021-12-28 00:58:20.965 T:4055120   DEBUG <general>: PLEX.subtitles: Number 1: Stream: {'id': '188775', 'streamType': '3', 'selected': '1', 'codec': 'pgs', 'index': '4', 'bitrate': '36', 'language': 'English', 'languageTag': 'en', 'languageCode': 'eng', 'title': 'Dialogue', 'displayTitle': 'English (PGS)', 'extendedDisplayTitle': 'Dialogue (English PGS)'}
2021-12-28 00:58:20.965 T:4055120   DEBUG <general>: PLEX.subtitles: Kodi list of external subs: []
...
2021-12-28 00:58:21.010 T:4055120   DEBUG <general>: PLEX.playlist_func: The PMS wants to display subtitle stream with Plex id 188775 and languageTag en
2021-12-28 00:58:21.010 T:4055120   DEBUG <general>: PLEX.playlist_func: Switching to Kodi subtitle stream number 1 because the PMS told us to show stream with Plex id 188775
2021-12-28 00:58:21.012 T:4055120   DEBUG <general>: PLEX.playlist_func: Switching subtitle stream
```
There is no Japanese, only 2 English tracks. 

Be aware that subtitles can be off on the Plex side, e.g. Plex thinks you have English subtitles but you might have something else!

**Comment by mason1171** [comment]
Thanks for the response.  If I’m reading that log snippet correctly, doesn’t it say that it’s switching to the second listed subtitle strain called Dialogue? That change is not reflecting in the video player. It keeps playing the file’s default subtitle stream. [Every episode is set like this.](https://user-images.githubusercontent.com/33766621/147609597-4fe9f65f-7fb1-46d0-9849-717229a5138b.jpeg) Each time, PKC plays one of these episodes it actually switches the Plex preferred subtitle to the default stream. (Signs/Songs). The Plex for kodi addon and plex app both switch properly. Tested on 2 devices with a few different shows. [Another log](https://we.tl/t-lFP9eAspBO). Can provide more if needed. I’ll try downgrading PKC and get back to you

**Comment by mason1171** [comment]
> Thanks for the response. If I’m reading that log snippet correctly, doesn’t it say that it’s switching to the second listed subtitle strain called Dialogue? That change is not reflecting in the video player. It keeps playing the file’s default subtitle stream. [Every episode is set like this.](https://user-images.githubusercontent.com/33766621/147609597-4fe9f65f-7fb1-46d0-9849-717229a5138b.jpeg) Each time, PKC plays one of these episodes it actually switches the Plex preferred subtitle to the default stream. (Signs/Songs). The Plex for kodi addon and plex app both switch properly. Tested on 2 devices with a few different shows. [Another log](https://we.tl/t-lFP9eAspBO). Can provide more if needed. I’ll try downgrading PKC and get back to you

Downgrading to 3.5.17 fixes it. This seems to be a 3.6.0+ issue
[Yet another log](https://we.tl/t-ELdD461Jtz)

**Comment by croneter** [comment]
Ough, embarrasing, really 🙈. Thanks for the bug report!! Fixed in PKC version 3.6.3

**Comment by mason1171** [comment]
> Ough, embarrasing, really 🙈. Thanks for the bug report!! Fixed in PKC version 3.6.3

Thanks for being an awesome dev. 🤝
